vscode/src/commands/context/shell.ts

```diff
@@ -21,6 +18,79 @@ Terminal output from the \`{command}\` command enclosed between <OUTPUT0412> tag
 {output}
 </OUTPUT0412>`
 
+// A persistent shell session that maintains state between commands
+class PersistentShell {
+    private shell: ReturnType<typeof spawn> | null = null
+    private buffer = ''
+
+    constructor() {
+        this.init()
+    }
+
+    private init() {
+        const shell = process.platform === 'win32' ? 'powershell.exe' : 'bash'
+        this.shell = spawn(shell, [], { stdio: ['pipe', 'pipe', 'pipe'] })
```

Comment:
```suggestion
        const cwd = vscode.workspace.workspaceFolders?.[0]?.uri?.fsPath
        this.shell = spawn(shell, [], {
            cwd,
            stdio: ['pipe', 'pipe', 'pipe'],
        })
```

or it defaults to the execution path of the extension host.